FHEM/00_SIGNALduino.pm

```diff
@@ -258,7 +258,7 @@ my %matchListSIGNALduino = (
 			'28:SD_Keeloq'				=> '^P(?:87|88)#.*',
 			'29:SD_GT'						=> '^P49#[A-Fa-f0-9]+',
 			'30:LaCrosse'					=> '^(\\S+\\s+9 |OK\\sWS\\s)',
-			'31:KOPP_FC'					=> '^kr.*',
+			'31:KOPP_FC'					=> '^kr..................',
```

Comment:
Was hat es denn damit auf sich dass Du so viele . da in die Regex setzt?

Reply:
Die vielen Punkte sind nichts anderes als eine "Verbesserung" des `.*` wo beliebig viele Zeichen folgen können.
Die Punkte geben die Zeichen vor aber eine andere Schreibweise wie kr gefolgt von 18 Zeichen habe ich nicht näher überarbeitet / angesehen.

Kurz gesagt, es wurde nur kopiert aber mit der Promisse, das wir somit vorgeben die Anzahl von Zeichen. Ja, es ist überarbeitunsfähig.

Reply:
Ich würde dir ja gerne ein Beispiel von regex101 schicken, aber vom Handy aus will das einfach nicht.

Das mit dem kopieren habe ich verstanden und ja ich habe es schon oft gesehen.
Leider habe ich da einen anderen Anspruch.

Es passt leider auch folgende Zeichenfolge:

kr84+€+;;(). Djueueubdierzhej8373-:34482

Um einfach nur bei einem groben Beispiel zu bleiben. Das einzigste was gegenüber .* Verändert wurde ist ja, die Mindest Anzahl an Zeichen. Ich habe da doch meine Zweifel ob das wirklich bewusst so gewollt ist.

Folgendes wäre equivalent dazu:
`^kr.{18,}`

Reply:
Ich habe mal soeben geschaut und werde das RegEx auf `^kr\w{18,}` ändern.
Das heißt, es dürfen nur Zahlen und Buchstaben, oder ein Unterstrich folgen mindest 18mal.
Somit würde ich nicht zulassen, das soetwas `kr84+€+;;(). Djueueubdi` passt.

Ich vermute ganz stark, das nur Zahlen und Buchstaben A-F gefolgt nach dem kr kommen aber da können wir nicht zu 100% wiederlegen. Aus diesem Grunde, denke ich mit `^kr\w{18,}` schon einen größeren Schritt der Einschränkung vorgenommen zu haben.